components/vision.py

```diff
@@ -69,83 +96,151 @@ def reproj(self) -> float:
     def using_multitag(self) -> bool:
         return self.has_multitag
 
+    @feedback
+    def raw_encoder_rotation(self) -> Rotation2d:
+        # The encoder has been set up to return values in the interval [0, 2pi]
+        return Rotation2d(self.encoder.get())
+
+    @feedback
+    def relative_bearing_to_closest_tag(self) -> Rotation2d:
+        # initialise default variables
+        # closest tag, distance, robot position
+        distance = math.inf
+        closest_bearing = Rotation2d()
+
+        for tag in APRILTAGS:
+            robot_to_tag = tag.pose.toPose2d() - self.chassis.get_pose()
+            relative_bearing = robot_to_tag.translation().angle()
+            if (
+                robot_to_tag.translation().norm() < distance
+                and -self.SERVO_HALF_ANGLE
+                <= relative_bearing.radians()
+                <= self.SERVO_HALF_ANGLE
```

Comment:
nit: this is a little bit difficult to read. Given the servo is intended to be centred around 0, I think this is easier to read?
```suggestion
                and abs(relative_bearing.radians()) <= self.SERVO_HALF_ANGLE
```